src/renderers/dom/client/wrappers/ReactDOMInput.js

```diff
@@ -214,17 +213,43 @@ var ReactDOMInput = {
       );
     }
 
+    var node = ReactDOMComponentTree.getNodeFromInstance(inst);
     var value = LinkedValueUtils.getValue(props);
     if (value != null) {
+
       // Cast `value` to a string to ensure the value is set correctly. While
       // browsers typically do this as necessary, jsdom doesn't.
-      DOMPropertyOperations.setValueForProperty(
-        ReactDOMComponentTree.getNodeFromInstance(inst),
-        'value',
-        '' + value
-      );
+      var newValue = '' + value;
+
+      // To avoid side effects (such as losing text selection), only set value if changed
+      if (newValue !== node.value) {
+        node.value = newValue;
+      }
+    } else {
+      if (props.value == null && props.defaultValue != null) {
+        invariant(inst._rootNodeID, 'Must be mounted to initialize initial input value');
+        node.defaultValue = '' + props.defaultValue;
+      }
+      if (props.checked == null && props.defaultChecked != null) {
+        invariant(inst._rootNodeID, 'Must be mounted to initialize initial input value');
+
+        // tmp is needed to work around a chrome bug where setting defaultChecked
+        // will sometimes influence the value of checked (even after detachment).
+        // Reference: https://bugs.chromium.org/p/chromium/issues/detail?id=608416
```

Comment:
Ok, I don't see how using tmp on update is any grosser/dirtier than using it on initial mount.  Especially since this means we incur unnecessary dom operations for every input instead of just updates to radio buttons.  But if you really like it better, I'll do it your way (unset name, set checked twice, set name) - updating diff now.